### PR TITLE
Include NGI role in revocation evaluation for orphaned API credentials

### LINKED ISSUE #366: Privacy work follow up
Following on from #284, here are some things to improve before the privacy restrictions can be activated:

- [x] If restricting personal data is in effect, one can only see personal data if one has a Site level role - not a NGI or Project level role. A NGI or Project level role should be sufficent to see personal data. Given https://github.com/GOCDB/gocdb/issues/364, I agree that currently a ServiceGroup role should not be sufficent to see personal data.
- [x] The NGI level contact info and user / role info should also be hidden if one doesn't have a role.
- [x] If one doesn't have a role, one can't see the "Request Role" button on a Site or Project entity (The only reason this doesn't apply to NGIs is the above point). The "Request Role" button should not be hidden.

As a stretch goal, or possibly as something to fix at a later date i.e. after privacy restrictions are activated. There's also a nuance whereby:
- If I have an NGI role, I can add an API identifier to a Site (so far so good)
- I may later gain a role over that site (okay...)
- I can't have that site role revoked while the API identifier is in place, even though I'd still have the NGI role (booo)

### PR DESCRIPTION
**Pull request by ineilson**
In combination with additions to #369 fixes #366 (final 3 bullets -  https://github.com/GOCDB/gocdb/issues/366#issue-1293298800)

### CODE REVIEW

File: htdocs/web_portal/views/user/view_user.php [hunk 1]
```diff
@@ -260,7 +260,7 @@
                     <?php
                         if(!$params['portalIsReadOnly'] &&
                             is_array($decorator) && $decorator["revokeButton"] == 'disabled') {
-                                echo 'Remove or reassign API credentials from site before revoking this role.';
+                                echo 'Remove or reassign API credentials from site(s) before revoking this role. Sites: ' . $decorator["revokeMessage"];
```
Comment (gregcorbett) [inline htdocs/web_portal/views/user/view_user.php:263]: Can this long line be broken up into shorter lines?

**Review by gregcorbett — CHANGES_REQUESTED** [review]
One comment so far about a long line.

Please also rebase this to resolve the conflict 😄

File: htdocs/web_portal/views/user/view_user.php (hunk 1 above)
Comment (ineilson) [reply htdocs/web_portal/views/user/view_user.php:263]: Clarification: you mean for the UI or 'just' in the code?

Comment (gregcorbett) [reply htdocs/web_portal/views/user/view_user.php:263]: 'just' in the code.

File: lib/Gocdb_Services/Role.php [hunk 2]
```diff
@@ -683,37 +699,60 @@ public function revokeRole(\Role $role, \User $callingUser){
...
     /**
      * Check that removing a site role from a user would not leave an orphaned API credential:
-     * attached to a site where the user no longer has a active role.
-     * @param \Role $role
-     * @throws \Exception If revoking the Role would leave an APIAuthentication
-     *                    credential owned by a user without a role at the site
+     * attached to a site where the user no longer has a active role at the site or owning NGI.
+     * @param \Role $role       Role to be revoked
+     * @return array            Array keyed by site name with values of credential count at named
+     *                          site blocking revocation
```
Comment (gregcorbett) [inline lib/Gocdb_Services/Role.php:705]: What do you mean by 'values of credential count at named site blocking revocation'? Do you mean the number of credentails at the site that are blocking the roles revokaiton?

If so, maybe reword the comment? It's taken me a while to parse what is going on.

**Review by gregcorbett — CHANGES_REQUESTED** [review]

**Comment by gregcorbett** [comment]
Also, the indenting in `checkOrphanAPIAuth` seemd a little weird to me - but I take it that's what the branch [psr-ngi-api-creds](https://github.com/ineilson/gocdb/tree/psr-ngi-api-creds) is for?

File: lib/Gocdb_Services/Role.php (hunk 2 above)
Comment (ineilson) [reply lib/Gocdb_Services/Role.php:705]: Yes, it took me a while too! Reworded.

File: htdocs/web_portal/views/user/view_user.php (hunk 1 above)
Comment (ineilson) [reply htdocs/web_portal/views/user/view_user.php]: Done.

**Comment by ineilson** [comment]
> Also, the indenting in `checkOrphanAPIAuth` seemd a little weird to me - but I take it that's what the branch [psr-ngi-api-creds](https://github.com/ineilson/gocdb/tree/psr-ngi-api-creds) is for?

Yes, in theory.

**Review by gregcorbett — APPROVED** [review]

**Comment by gregcorbett** [comment]
Just rebasing this on top of the latest dev before merging.

**Comment by gregcorbett** [comment]
> > Also, the indenting in `checkOrphanAPIAuth` seemd a little weird to me - but I take it that's what the branch [psr-ngi-api-creds](https://github.com/ineilson/gocdb/tree/psr-ngi-api-creds) is for?
> 
> Yes, in theory.

I look forward to the PR 😄

**Comment by gregcorbett** [comment]
> I look forward to the PR 😄

Ah I see it already exists #372 😋
